### PR TITLE
"Town portal" spell window should have evil UI elements

### LINKED ISSUE #2240: "Town portal" spell window should have evil UI elements
Buttons, scroll bar and list background in evil interface are the same as for good interface.
image

### PR DESCRIPTION
**Pull request by shprotru**
Fix #2240

### CODE REVIEW

File: src/fheroes2/agg/icn.h [hunk 1]
```diff
@@ -916,7 +916,9 @@ namespace ICN
 
         TROLL2MSL,
```
Comment (ihhub) [inline src/fheroes2/agg/icn.h:918]: Please remove this empty line.

File: src/fheroes2/heroes/heroes_spell.cpp [hunk 2]
```diff
@@ -125,34 +126,33 @@ void CastleIndexListBox::RedrawBackground( const Point & dst )
     text.Set( _( "Select town to port to." ), Font::BIG );
     text.Blit( dst.x + 140 - text.w() / 2, dst.y + 25 );
 
-    int32_t totalHeight = rtAreaItems.height + 6;
-
     int32_t offsetY = 45;
-    const fheroes2::Sprite & upperPart = fheroes2::AGG::GetICN( ICN::LISTBOX, 0 );
-    const fheroes2::Sprite & middlePart = fheroes2::AGG::GetICN( ICN::LISTBOX, 1 );
-    const fheroes2::Sprite & lowerPart = fheroes2::AGG::GetICN( ICN::LISTBOX, 2 );
+    const fheroes2::Sprite & upperPart = fheroes2::AGG::GetICN( conf.ExtGameEvilInterface() ? ICN::LISTBOX_EVIL : ICN::LISTBOX, 0 );
```
Comment (ihhub) [inline src/fheroes2/heroes/heroes_spell.cpp:130]: Please cache `const int listId = conf.ExtGameEvilInterface() ? ICN::LISTBOX_EVIL : ICN::LISTBOX;`.

Comment (ihhub) [reply src/fheroes2/heroes/heroes_spell.cpp:130]: Also please cache `conf.ExtGameEvilInterface()` as well.

File: src/fheroes2/heroes/heroes_spell.cpp [hunk 3]
```diff
@@ -523,18 +524,21 @@ bool ActionSpellTownPortal( Heroes & hero )
...
     listbox.SetAreaMaxItems( 5 );
     listbox.SetAreaItems( fheroes2::Rect( area.x + 6, area.y + 49, 250, 160 ) );
     listbox.SetListContent( castles );
     listbox.RedrawBackground( area );
     listbox.Redraw();
 
     fheroes2::ButtonGroup btnGroups;
-    btnGroups.createButton( area.x, area.y + 222, ICN::REQUEST, 1, 2, Dialog::OK );
-    btnGroups.createButton( area.x + 182, area.y + 222, ICN::REQUEST, 3, 4, Dialog::CANCEL );
+    const int btnICNID = conf.ExtGameEvilInterface() ? ICN::SYSTEME : ICN::SYSTEM;
```
Comment (ihhub) [inline src/fheroes2/heroes/heroes_spell.cpp:538]: Please name it as `buttonIcnId` for just easier reading.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @shprotru I left few comment. Please take a look.

File: src/fheroes2/agg/icn.h (hunk 1 above)
Comment (shprotru) [reply src/fheroes2/agg/icn.h]: done

File: src/fheroes2/heroes/heroes_spell.cpp (hunk 2 above)
Comment (shprotru) [reply src/fheroes2/heroes/heroes_spell.cpp]: done

File: src/fheroes2/heroes/heroes_spell.cpp (hunk 3 above)
Comment (shprotru) [reply src/fheroes2/heroes/heroes_spell.cpp]: done

File: src/fheroes2/heroes/heroes_spell.cpp [hunk 4]
```diff
@@ -118,41 +118,42 @@ void CastleIndexListBox::RedrawItem( const s32 & index, s32 dstx, s32 dsty, bool
...
     int32_t offsetY = 45;
-    const fheroes2::Sprite & upperPart = fheroes2::AGG::GetICN( ICN::LISTBOX, 0 );
-    const fheroes2::Sprite & middlePart = fheroes2::AGG::GetICN( ICN::LISTBOX, 1 );
-    const fheroes2::Sprite & lowerPart = fheroes2::AGG::GetICN( ICN::LISTBOX, 2 );
-
     fheroes2::Blit( upperPart, display, dst.x + 2, dst.y + offsetY );
 
     offsetY += upperPart.height();
 
+    int32_t totalHeight = rtAreaItems.height + 6;
```
Comment (ihhub) [inline src/fheroes2/heroes/heroes_spell.cpp:139]: Could you please explain why do we need this change?

Comment (ihhub) [reply src/fheroes2/heroes/heroes_spell.cpp:139]: Ah, you just moved it below.

Comment (shprotru) [reply src/fheroes2/heroes/heroes_spell.cpp:139]: yeah, `totalHeight` moved closer to place where it's used

**Comment by ihhub** [comment]
@shprotru thank you very much for fixing this issue!

**Pull request merged**